src/util/discord.ts

```diff
@@ -40,3 +40,28 @@ export const createDiscordEvent = async (
 
   return response;
 };
+
+export const addCoverEvent = async (
+  eventID: string,
+  image: any,
+) => {
+  // Discord takes image in format: data:image/<extension>;base64,BASE64_ENCODED_IMAGE_DATA
+  const imageUrlToBase64 = async (url: string) => {
+    const data = await fetch(url);
+    const blob = await data.arrayBuffer()
+    const extRe = new RegExp("\.([0-9a-z]+)(?:[\?#]|$)", "i");
+    const ext = extRe.exec(image)[1].toString().toLowerCase();
+
+    return `data:image/${ext};base64,${Buffer.from(blob).toString('base64')}`
+  };
+
+  var imageURI = (await imageUrlToBase64(image)).toString();
+
+  const rest = new REST({ version: "10" }).setToken(process.env.DISCORD_BOT_TOKEN as string);
+  const response = await rest.patch(`/guilds/${process.env.ACM_GUILD_ID}/scheduled-events/${eventID}`, {
+      body: { image: imageURI, },
+    }
+  );
```

Comment:
Instead of doing a PATCH here, can we just upload the details and the image at the same time in the same POST request?
https://discord.com/developers/docs/resources/guild-scheduled-event#create-guild-scheduled-event allows for an image param